util/chplenv/chpl_atomics.py

```diff
@@ -10,13 +10,21 @@
 @memoize
 def get(flag='target'):
     if flag == 'network':
+        valid = ['none', 'ofi', 'ugni']
         atomics_val = overrides.get('CHPL_NETWORK_ATOMICS')
+        comm_val = chpl_comm.get()
         if not atomics_val:
-            comm_val = chpl_comm.get()
             if comm_val in ['ofi', 'ugni'] and get('target') != 'locks':
                 atomics_val = comm_val
             else:
                 atomics_val = 'none'
+        elif atomics_val == 'gasnet':
+            error("CHPL_NETWORK_ATOMICS=gasnet is not supported")
+        elif atomics_val not in valid:
+            error("CHPL_NETWORK_ATOMICS must be one of " + ','.join(valid))
```

Comment:
this looks good to me!

I think arguably now the special case for gasnet isn't really needed, but it does give it a nicer error message, so I'm all for it

Reply:
Yes, I wanted to give a more specific error message for gasnet.